alvr/dashboard/src/steamvr_launcher/linux_steamvr.rs

```diff
@@ -76,59 +76,124 @@ pub fn linux_hardware_checks() {
             adapter.get_info().device_type == wgpu::DeviceType::DiscreteGpu
                 || adapter.get_info().device_type == wgpu::DeviceType::IntegratedGpu
         })
-        .map(|adapter| match adapter.get_info().vendor {
-            0x10de => DeviceInfo::Nvidia,
-            0x1002 => DeviceInfo::Amd {
-                device_type: adapter.get_info().device_type,
-            },
-            0x8086 => DeviceInfo::Intel {
-                device_type: adapter.get_info().device_type,
-            },
-            _ => DeviceInfo::Unknown,
+        .map(|adapter| {
+            let vendor = match adapter.get_info().vendor {
+                0x10de => DeviceInfo::Nvidia,
+                0x1002 => DeviceInfo::Amd {
+                    device_type: adapter.get_info().device_type,
+                },
+                0x8086 => DeviceInfo::Intel {
+                    device_type: adapter.get_info().device_type,
+                },
+                _ => DeviceInfo::Unknown,
+            };
+
+            (adapter, vendor)
         })
         .collect::<Vec<_>>();
-    linux_hybrid_gpu_checks(&device_infos);
+    linux_gpu_checks(&device_infos);
     linux_encoder_checks(&device_infos);
 }
 
-fn linux_hybrid_gpu_checks(device_infos: &[DeviceInfo]) {
+fn linux_gpu_checks(device_infos: &[(&wgpu::Adapter, DeviceInfo)]) {
     let have_igpu = device_infos.iter().any(|gpu| {
-        gpu == &DeviceInfo::Amd {
-            device_type: wgpu::DeviceType::IntegratedGpu,
-        } || gpu
-            == &DeviceInfo::Intel {
+        gpu.1
+            == DeviceInfo::Amd {
                 device_type: wgpu::DeviceType::IntegratedGpu,
             }
+            || gpu.1
+                == DeviceInfo::Intel {
+                    device_type: wgpu::DeviceType::IntegratedGpu,
+                }
     });
     debug!("have_igpu: {}", have_igpu);
-    let have_nvidia_dgpu = device_infos.iter().any(|gpu| gpu == &DeviceInfo::Nvidia);
+
+    let have_nvidia_dgpu = device_infos.iter().any(|gpu| gpu.1 == DeviceInfo::Nvidia);
     debug!("have_nvidia_dgpu: {}", have_nvidia_dgpu);
+
+    let have_amd_igpu = device_infos.iter().any(|gpu| {
+        gpu.1
+            == DeviceInfo::Amd {
+                device_type: wgpu::DeviceType::IntegratedGpu,
+            }
+    });
+    debug!("have_amd_igpu: {}", have_amd_igpu);
```

Comment:
We shouldn't be checking for the integrated gpu thing twice. 
We should have the `have_amd_igpu` and `have_intel_igpu` at the top and just have `have_igpu` be an or of them.